signer/aws/aws.go

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// The OpenSearch Contributors require contributions made to
+// this file be licensed under the Apache-2.0 license or a
+// compatible open source license.
+//
+// Modifications Copyright OpenSearch Contributors. See
+// GitHub history for details.
+
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+//OpenSearchService AWS OpenSearchService Name
+const OpenSearchService = "es"
+
+// Signer is a interface that will implement opensearchtransport.Signer
+type Signer struct {
+	session session.Session
+	service string
+}
+
+//NewSigner returns an instance of Signer
```

Comment:
Nit: space after `//`, there's a few others like this.

Reply:
Ack